Fix 2 broken tests.

A couple of the tests are broken, diagnose the issues and fix them.

```bash
scheduling tests via LoadScheduling

tests/test_hash_model.py::test_full_text_search_queries
tests/test_hash_model.py::test_exact_match_queries
tests/test_hash_model.py::test_recursive_query_resolution
tests/test_hash_model.py::test_tag_queries_punctuation
tests/test_hash_model.py::test_tag_queries_negation
tests/test_hash_model.py::test_tag_queries_boolean_logic
tests/test_hash_model.py::test_numeric_queries
tests/test_hash_model.py::test_validates_required_fields
tests/test_hash_model.py::test_sorting
tests/test_hash_model.py::test_validates_field
tests/test_hash_model.py::test_validation_passes
tests/test_hash_model.py::test_raises_error_with_dataclasses
tests/test_hash_model.py::test_raises_error_with_sets
tests/test_hash_model.py::test_raises_error_with_dicts
tests/test_hash_model.py::test_saves_model_and_creates_pk
tests/test_hash_model.py::test_raises_error_with_embedded_models
[gw10] [  0%] PASSED tests/test_hash_model.py::test_validation_passes
[gw8] [  1%] PASSED tests/test_hash_model.py::test_validates_required_fields
[gw9] [  2%] PASSED tests/test_hash_model.py::test_validates_field
tests/test_json_model.py::test_saves_model_and_creates_pk
[gw14] [  3%] PASSED tests/test_hash_model.py::test_raises_error_with_dicts
[gw15] [  4%] PASSED tests/test_hash_model.py::test_raises_error_with_sets
tests/test_json_model.py::test_validates_field
[gw12] [  4%] PASSED tests/test_hash_model.py::test_raises_error_with_embedded_models
[gw13] [  5%] PASSED tests/test_hash_model.py::test_raises_error_with_dataclasses
tests/test_json_model.py::test_validation_passes
tests/test_json_model.py::test_access_result_by_index_cached
tests/test_json_model.py::test_saves_many_explicit_transaction
tests/test_json_model.py::test_paginate_query
[gw11] [  6%] PASSED tests/test_hash_model.py::test_saves_model_and_creates_pk
tests/test_json_model.py::test_updates_a_model
tests/test_json_model.py::test_saves_many_implicit_pipeline
[gw3] [  7%] PASSED tests/test_hash_model.py::test_tag_queries_boolean_logic
[gw2] [  8%] PASSED tests/test_hash_model.py::test_recursive_query_resolution
[gw1] [  9%] PASSED tests/test_hash_model.py::test_full_text_search_queries
tests/test_hash_model.py::test_paginate_query
tests/test_hash_model.py::test_saves_many
tests/test_hash_model.py::test_updates_a_model
[gw4] [  9%] PASSED tests/test_hash_model.py::test_tag_queries_punctuation
[gw7] [ 10%] PASSED tests/test_hash_model.py::test_sorting
tests/test_hash_model.py::test_access_result_by_index_cached
tests/test_json_model.py::test_validates_required_fields
[gw5] [ 11%] PASSED tests/test_hash_model.py::test_tag_queries_negation
[gw0] [ 12%] PASSED tests/test_hash_model.py::test_exact_match_queries
tests/test_hash_model.py::test_access_result_by_index_not_cached
tests/test_hash_model.py::test_raises_error_with_lists
[gw6] [ 13%] PASSED tests/test_hash_model.py::test_numeric_queries
tests/test_hash_model.py::test_schema
[gw8] [ 13%] PASSED tests/test_json_model.py::test_validates_field
[gw9] [ 14%] PASSED tests/test_json_model.py::test_validation_passes
tests/test_json_model.py::test_in_query
tests/test_json_model.py::test_update_query
[gw1] [ 15%] PASSED tests/test_hash_model.py::test_saves_many
[gw11] [ 16%] PASSED tests/test_json_model.py::test_saves_many_implicit_pipeline
tests/test_json_model.py::test_tag_queries_negation
[gw10] [ 17%] PASSED tests/test_json_model.py::test_saves_model_and_creates_pk
[gw0] [ 18%] PASSED tests/test_hash_model.py::test_raises_error_with_lists
[gw7] [ 18%] PASSED tests/test_json_model.py::test_validates_required_fields
[gw3] [ 19%] PASSED tests/test_hash_model.py::test_paginate_query
tests/test_json_model.py::test_tag_queries_boolean_logic
[gw4] [ 20%] PASSED tests/test_hash_model.py::test_access_result_by_index_cached
[gw2] [ 21%] PASSED tests/test_hash_model.py::test_updates_a_model
tests/test_json_model.py::test_access_result_by_index_not_cached
[gw15] [ 22%] PASSED tests/test_json_model.py::test_access_result_by_index_cached
tests/test_json_model.py::test_not_found
[gw6] [ 22%] PASSED tests/test_hash_model.py::test_schema
tests/test_json_model.py::test_allows_dataclasses
tests/test_json_model.py::test_sorting
tests/test_json_model.py::test_numeric_queries
tests/test_json_model.py::test_tag_queries_punctuation
tests/test_json_model.py::test_exact_match_queries
tests/test_json_model.py::test_allows_and_serializes_dicts
[gw14] [ 23%] PASSED tests/test_json_model.py::test_paginate_query
[gw12] [ 24%] PASSED tests/test_json_model.py::test_saves_many_explicit_transaction
[gw13] [ 25%] PASSED tests/test_json_model.py::test_updates_a_model
[gw5] [ 26%] PASSED tests/test_hash_model.py::test_access_result_by_index_not_cached
tests/test_json_model.py::test_recursive_query_expression_resolution
tests/test_json_model.py::test_recursive_query_field_resolution
tests/test_json_model.py::test_full_text_search
tests/test_json_model.py::test_list_field_limitations
[gw7] [ 27%] PASSED tests/test_json_model.py::test_not_found
tests/test_oss_redis_features.py::test_validates_required_fields
[gw11] [ 27%] PASSED tests/test_json_model.py::test_tag_queries_boolean_logic
tests/test_oss_redis_features.py::test_all_keys
[gw8] [ 28%] PASSED tests/test_json_model.py::test_in_query
[gw0] [ 29%] PASSED tests/test_json_model.py::test_allows_dataclasses
[gw9] [ 30%] PASSED tests/test_json_model.py::test_update_query
tests/test_json_model.py::test_allows_and_serializes_sets
tests/test_oss_redis_features.py::test_validates_field
tests/test_json_model.py::test_allows_and_serializes_lists
[gw14] [ 31%] PASSED tests/test_json_model.py::test_recursive_query_expression_resolution
[gw6] [ 31%] PASSED tests/test_json_model.py::test_allows_and_serializes_dicts
[gw1] [ 32%] PASSED tests/test_json_model.py::test_tag_queries_negation
tests_sync/test_hash_model.py::test_exact_match_queries
[gw3] [ 33%] PASSED tests/test_json_model.py::test_tag_queries_punctuation
tests/test_oss_redis_features.py::test_updates_a_model
tests/test_json_model.py::test_schema
tests/test_oss_redis_features.py::test_raises_error_with_embedded_models
[gw10] [ 34%] PASSED tests/test_json_model.py::test_access_result_by_index_not_cached
tests/test_oss_redis_features.py::test_not_found
[gw7] [ 35%] PASSED tests/test_oss_redis_features.py::test_validates_required_fields
tests_sync/test_hash_model.py::test_tag_queries_boolean_logic
[gw2] [ 36%] PASSED tests/test_json_model.py::test_numeric_queries
tests/test_oss_redis_features.py::test_saves_model_and_creates_pk
[gw4] [ 36%] PASSED tests/test_json_model.py::test_sorting
tests/test_oss_redis_features.py::test_validation_passes
[gw9] [ 37%] PASSED tests/test_json_model.py::test_allows_and_serializes_lists
[gw8] [ 38%] PASSED tests/test_json_model.py::test_allows_and_serializes_sets
[gw15] [ 39%] PASSED tests/test_json_model.py::test_exact_match_queries
tests_sync/test_hash_model.py::test_sorting
tests_sync/test_hash_model.py::test_numeric_queries
[gw13] [ 40%] PASSED tests/test_json_model.py::test_full_text_search
[gw12] [ 40%] PASSED tests/test_json_model.py::test_recursive_query_field_resolution
tests/test_oss_redis_features.py::test_saves_many
[gw0] [ 41%] PASSED tests/test_oss_redis_features.py::test_validates_field
tests_sync/test_hash_model.py::test_full_text_search_queries
tests/test_pydantic_integrations.py::test_email_str
tests_sync/test_hash_model.py::test_tag_queries_negation
[gw1] [ 42%] PASSED tests/test_json_model.py::test_schema
tests_sync/test_hash_model.py::test_validation_passes
[gw14] [ 43%] PASSED tests_sync/test_hash_model.py::test_exact_match_queries
[gw3] [ 44%] PASSED tests/test_oss_redis_features.py::test_raises_error_with_embedded_models
[gw10] [ 45%] PASSED tests/test_oss_redis_features.py::test_not_found
tests_sync/test_hash_model.py::test_validates_required_fields
[gw5] [ 45%] PASSED tests/test_json_model.py::test_list_field_limitations
tests_sync/test_hash_model.py::test_saves_model_and_creates_pk
tests_sync/test_hash_model.py::test_raises_error_with_embedded_models
[gw7] [ 46%] PASSED tests_sync/test_hash_model.py::test_tag_queries_boolean_logic
tests_sync/test_hash_model.py::test_recursive_query_resolution
tests_sync/test_hash_model.py::test_raises_error_with_dataclasses
[gw11] [ 47%] PASSED tests/test_oss_redis_features.py::test_all_keys
[gw2] [ 48%] PASSED tests/test_oss_redis_features.py::test_saves_model_and_creates_pk
tests_sync/test_hash_model.py::test_raises_error_with_dicts
tests_sync/test_hash_model.py::test_tag_queries_punctuation
[gw4] [ 49%] PASSED tests/test_oss_redis_features.py::test_validation_passes
tests_sync/test_hash_model.py::test_raises_error_with_sets
[gw6] [ 50%] PASSED tests/test_oss_redis_features.py::test_updates_a_model
tests_sync/test_hash_model.py::test_validates_field
[gw1] [ 50%] PASSED tests_sync/test_hash_model.py::test_validation_passes
[gw9] [ 51%] PASSED tests_sync/test_hash_model.py::test_sorting
tests_sync/test_hash_model.py::test_schema
tests_sync/test_hash_model.py::test_raises_error_with_lists
[gw14] [ 52%] PASSED tests_sync/test_hash_model.py::test_validates_required_fields
tests_sync/test_json_model.py::test_validates_required_fields
[gw15] [ 53%] PASSED tests/test_oss_redis_features.py::test_saves_many
[gw7] [ 54%] PASSED tests_sync/test_hash_model.py::test_raises_error_with_dataclasses
[gw10] [ 54%] PASSED tests_sync/test_hash_model.py::test_raises_error_with_embedded_models
[gw3] [ 55%] PASSED tests_sync/test_hash_model.py::test_saves_model_and_creates_pk
tests_sync/test_hash_model.py::test_updates_a_model
[gw0] [ 56%] PASSED tests_sync/test_hash_model.py::test_tag_queries_negation
[gw2] [ 57%] PASSED tests_sync/test_hash_model.py::test_raises_error_with_dicts
tests_sync/test_json_model.py::test_saves_many_implicit_pipeline
tests_sync/test_json_model.py::test_validation_passes
tests_sync/test_json_model.py::test_validates_field
[gw4] [ 58%] PASSED tests_sync/test_hash_model.py::test_raises_error_with_sets
tests_sync/test_hash_model.py::test_access_result_by_index_not_cached
tests_sync/test_json_model.py::test_saves_many_explicit_transaction
tests_sync/test_json_model.py::test_paginate_query
[gw5] [ 59%] PASSED tests_sync/test_hash_model.py::test_recursive_query_resolution
tests_sync/test_json_model.py::test_saves_model_and_creates_pk
[gw6] [ 59%] PASSED tests_sync/test_hash_model.py::test_validates_field
[gw1] [ 60%] PASSED tests_sync/test_hash_model.py::test_schema
tests_sync/test_json_model.py::test_access_result_by_index_cached
[gw11] [ 61%] PASSED tests_sync/test_hash_model.py::test_tag_queries_punctuation
[gw9] [ 62%] PASSED tests_sync/test_hash_model.py::test_raises_error_with_lists
tests_sync/test_json_model.py::test_access_result_by_index_not_cached
tests_sync/test_json_model.py::test_updates_a_model
tests_sync/test_json_model.py::test_in_query
[gw14] [ 63%] PASSED tests_sync/test_json_model.py::test_validates_required_fields
tests_sync/test_json_model.py::test_update_query
[gw3] [ 63%] PASSED tests_sync/test_json_model.py::test_validates_field
tests_sync/test_json_model.py::test_full_text_search
[gw13] [ 64%] FAILED tests_sync/test_hash_model.py::test_full_text_search_queries
[gw7] [ 65%] PASSED tests_sync/test_json_model.py::test_saves_many_implicit_pipeline
[gw8] [ 66%] FAILED tests_sync/test_hash_model.py::test_numeric_queries
tests_sync/test_hash_model.py::test_paginate_query
tests_sync/test_json_model.py::test_recursive_query_expression_resolution
[gw15] [ 67%] PASSED tests_sync/test_hash_model.py::test_updates_a_model
tests_sync/test_hash_model.py::test_saves_many
[gw10] [ 68%] PASSED tests_sync/test_json_model.py::test_validation_passes
tests_sync/test_json_model.py::test_exact_match_queries
tests_sync/test_json_model.py::test_recursive_query_field_resolution
[gw2] [ 68%] PASSED tests_sync/test_json_model.py::test_saves_many_explicit_transaction
tests_sync/test_json_model.py::test_tag_queries_punctuation
[gw0] [ 69%] PASSED tests_sync/test_hash_model.py::test_access_result_by_index_not_cached
tests_sync/test_json_model.py::test_tag_queries_boolean_logic
[gw4] [ 70%] PASSED tests_sync/test_json_model.py::test_paginate_query
[gw5] [ 71%] PASSED tests_sync/test_json_model.py::test_saves_model_and_creates_pk
tests_sync/test_json_model.py::test_tag_queries_negation
tests_sync/test_json_model.py::test_numeric_queries
[gw9] [ 72%] PASSED tests_sync/test_json_model.py::test_in_query
[gw6] [ 72%] PASSED tests_sync/test_json_model.py::test_access_result_by_index_cached
tests_sync/test_json_model.py::test_allows_dataclasses
tests_sync/test_json_model.py::test_sorting
[gw8] [ 73%] PASSED tests_sync/test_hash_model.py::test_saves_many
[gw14] [ 74%] PASSED tests_sync/test_json_model.py::test_update_query
tests_sync/test_oss_redis_features.py::test_all_keys
tests_sync/test_json_model.py::test_allows_and_serializes_dicts
[gw11] [ 75%] PASSED tests_sync/test_json_model.py::test_updates_a_model
tests_sync/test_json_model.py::test_list_field_limitations
[gw1] [ 76%] PASSED tests_sync/test_json_model.py::test_access_result_by_index_not_cached
[gw13] [ 77%] PASSED tests_sync/test_hash_model.py::test_paginate_query
tests_sync/test_json_model.py::test_not_found
[gw7] [ 77%] PASSED tests_sync/test_json_model.py::test_recursive_query_expression_resolution
tests_sync/test_json_model.py::test_allows_and_serializes_lists
[gw3] [ 78%] PASSED tests_sync/test_json_model.py::test_full_text_search
tests_sync/test_json_model.py::test_schema
tests_sync/test_json_model.py::test_allows_and_serializes_sets
[gw9] [ 79%] PASSED tests_sync/test_json_model.py::test_allows_dataclasses
[gw0] [ 80%] PASSED tests_sync/test_json_model.py::test_tag_queries_boolean_logic
tests_sync/test_oss_redis_features.py::test_saves_many
[gw2] [ 81%] PASSED tests_sync/test_json_model.py::test_tag_queries_punctuation
tests_sync/test_oss_redis_features.py::test_validation_passes
tests_sync/test_oss_redis_features.py::test_validates_field
[gw10] [ 81%] PASSED tests_sync/test_json_model.py::test_recursive_query_field_resolution
tests_sync/test_oss_redis_features.py::test_validates_required_fields
[gw6] [ 82%] PASSED tests_sync/test_json_model.py::test_sorting
[gw15] [ 83%] PASSED tests_sync/test_json_model.py::test_exact_match_queries
tests_sync/test_oss_redis_features.py::test_not_found
tests_sync/test_oss_redis_features.py::test_updates_a_model
[gw4] [ 84%] PASSED tests_sync/test_json_model.py::test_tag_queries_negation
tests_sync/test_oss_redis_features.py::test_saves_model_and_creates_pk
[gw5] [ 85%] PASSED tests_sync/test_json_model.py::test_numeric_queries
[gw14] [ 86%] PASSED tests_sync/test_json_model.py::test_allows_and_serializes_dicts
tests_sync/test_oss_redis_features.py::test_raises_error_with_embedded_models
[gw1] [ 86%] PASSED tests_sync/test_json_model.py::test_not_found
[gw7] [ 87%] PASSED tests_sync/test_json_model.py::test_schema
[gw13] [ 88%] PASSED tests_sync/test_json_model.py::test_allows_and_serializes_lists
[gw3] [ 89%] PASSED tests_sync/test_json_model.py::test_allows_and_serializes_sets
[gw0] [ 90%] PASSED tests_sync/test_oss_redis_features.py::test_validation_passes
[gw2] [ 90%] PASSED tests_sync/test_oss_redis_features.py::test_validates_field
[gw10] [ 91%] PASSED tests_sync/test_oss_redis_features.py::test_validates_required_fields
[gw11] [ 92%] PASSED tests_sync/test_json_model.py::test_list_field_limitations
[gw15] [ 93%] PASSED tests_sync/test_oss_redis_features.py::test_not_found
[gw9] [ 94%] PASSED tests_sync/test_oss_redis_features.py::test_saves_many
[gw6] [ 95%] PASSED tests_sync/test_oss_redis_features.py::test_updates_a_model
[gw5] [ 95%] PASSED tests_sync/test_oss_redis_features.py::test_raises_error_with_embedded_models
[gw4] [ 96%] PASSED tests_sync/test_oss_redis_features.py::test_saves_model_and_creates_pk
[gw8] [ 97%] PASSED tests_sync/test_oss_redis_features.py::test_all_keys
tests_sync/test_pydantic_integrations.py::test_email_str
[gw12] [ 98%] PASSED tests/test_pydantic_integrations.py::test_email_str
tests_sync/test_hash_model.py::test_access_result_by_index_cached
[gw12] [ 99%] PASSED tests_sync/test_hash_model.py::test_access_result_by_index_cached
[gw8] [100%] PASSED tests_sync/test_pydantic_integrations.py::test_email_str

============================================ FAILURES ============================================
_________________________________ test_full_text_search_queries __________________________________
[gw13] darwin -- Python 3.9.5 /Users/simonprickett/Library/Caches/pypoetry/virtualenvs/redis-om-dXPLVk-w-py3.9/bin/python

members = (Member(pk='01FS5ATV1SMS7T8PG5MJM4PB7P', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=d...com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.'))
m = Models(BaseHashModel=<class 'tests_sync.test_hash_model.m.<locals>.BaseHashModel'>, Order=<class 'tests_sync.test_hash_model.m.<locals>.Order'>, Member=<class 'tests_sync.test_hash_model.m.<locals>.Member'>)

    @pytest.mark.asyncio
    def test_full_text_search_queries(members, m):
        member1, member2, member3 = members

        actual = (
            m.Member.find(m.Member.bio % "great").all()
        )

        assert actual == [member1]

        actual = (
            m.Member.find(~(m.Member.bio % "anxious")).all()
        )

>       assert actual == [member1, member3]
E       AssertionError: assert [Member(pk='01FS5ATV1SEMK9FDCVQ4075VPR', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.'),\n Member(pk='01FS5ATV1SMS7T8PG5MJM4PB7P', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.')] == [Member(pk='01FS5ATV1SMS7T8PG5MJM4PB7P', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.'),\n Member(pk='01FS5ATV1SEMK9FDCVQ4075VPR', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.')]
E         At index 0 diff: Member(pk='01FS5ATV1SEMK9FDCVQ4075VPR', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.') != Member(pk='01FS5ATV1SMS7T8PG5MJM4PB7P', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.')
E         Full diff:
E           [
E         +  Member(pk='01FS5ATV1SEMK9FDCVQ4075VPR', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.'),
E            Member(pk='01FS5ATV1SMS7T8PG5MJM4PB7P', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.'),
E         -  Member(pk='01FS5ATV1SEMK9FDCVQ4075VPR', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.'),
E           ]

tests_sync/test_hash_model.py:145: AssertionError
______________________________________ test_numeric_queries ______________________________________
[gw8] darwin -- Python 3.9.5 /Users/simonprickett/Library/Caches/pypoetry/virtualenvs/redis-om-dXPLVk-w-py3.9/bin/python

members = (Member(pk='01FS5ATV1DNY2P4WKGYVMZKCXF', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=d...com', join_date=datetime.date(2022, 1, 11), age=100, bio='This is member 3 who is a funny and lively sort of person.'))
m = Models(BaseHashModel=<class 'tests_sync.test_hash_model.m.<locals>.BaseHashModel'>, Order=<class 'tests_sync.test_hash_model.m.<locals>.Order'>, Member=<class 'tests_sync.test_hash_model.m.<locals>.Member'>)

    @pytest.mark.asyncio
    def test_numeric_queries(members, m):
        member1, member2, member3 = members

        actual = m.Member.find(m.Member.age == 34).all()
        assert actual == [member2]

        actual = m.Member.find(m.Member.age > 34).sort_by("age").all()
        assert actual == [member1, member3]

        actual = m.Member.find(m.Member.age < 35).all()
        assert actual == [member2]

        actual = m.Member.find(m.Member.age <= 34).all()
        assert actual == [member2]

        actual = m.Member.find(m.Member.age >= 100).all()
        assert actual == [member3]

        actual = m.Member.find(m.Member.age != 34).sort_by("age").all()
        assert actual == [member1, member3]

        actual = m.Member.find(~(m.Member.age == 100)).sort_by("age").all()
>       assert actual == [member2, member1]
E       AssertionError: assert [Member(pk='01FS5ATV1D5SWT66T89028YN2X', first_name='Kim', last_name='Brookins', email='k@example.com', join_date=datetime.date(2022, 1, 11), age=34, bio='This is member 2 who can be quite anxious until you get to know them.')] == [Member(pk='01FS5ATV1D5SWT66T89028YN2X', first_name='Kim', last_name='Brookins', email='k@example.com', join_date=datetime.date(2022, 1, 11), age=34, bio='This is member 2 who can be quite anxious until you get to know them.'),\n Member(pk='01FS5ATV1DNY2P4WKGYVMZKCXF', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.')]
E         Right contains one more item: Member(pk='01FS5ATV1DNY2P4WKGYVMZKCXF', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=da...date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.')
E         Full diff:
E           [
E            Member(pk='01FS5ATV1D5SWT66T89028YN2X', first_name='Kim', last_name='Brookins', email='k@example.com', join_date=datetime.date(2022, 1, 11), age=34, bio='This is member 2 who can be quite anxious until you get to know them.'),
E         -  Member(pk='01FS5ATV1DNY2P4WKGYVMZKCXF', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=datetime.date(2022, 1, 11), age=38, bio='This is member 1 whose greatness makes him the life and soul of any party he goes to.'),
E           ]

tests_sync/test_hash_model.py:310: AssertionError

---------- coverage: platform darwin, python 3.9.5-final-0 -----------
Name                                     Stmts   Miss  Cover   Missing
----------------------------------------------------------------------
aredis_om/__init__.py                        4      0   100%
aredis_om/checks.py                         21     12    43%   9-10, 14-17, 21-26
aredis_om/connections.py                    12      3    75%   20-22
aredis_om/model/__init__.py                  2      0   100%
aredis_om/model/cli/__init__.py              0      0   100%
aredis_om/model/cli/migrate.py              13     13     0%   1-18
aredis_om/model/encoders.py                 72     35    51%   68, 70, 73-86, 94, 96, 98, 132-147, 150-155, 159-173
aredis_om/model/migrations/__init__.py       0      0   100%
aredis_om/model/migrations/migrator.py      84     14    83%   24-35, 49, 76-77, 82-83, 94, 105-107
aredis_om/model/model.py                   857    118    86%   97, 108, 125, 133, 142-149, 163, 182, 190, 196, 200, 204, 208-211, 215, 238, 242, 294, 302, 348, 388, 395, 413, 440, 468, 493, 496-502, 521, 523, 527, 555-565, 583-586, 597, 644, 658-663, 676, 690, 692, 694, 696, 754, 765, 795, 798-803, 819-829, 879, 902-903, 1047, 1110, 1118, 1122, 1126, 1129, 1145, 1147, 1178-1181, 1261, 1267, 1327-1335, 1349, 1380, 1389-1398, 1402, 1417-1425, 1436-1446, 1459, 1543-1544, 1571-1574, 1658, 1662-1666
aredis_om/model/query_resolver.py           23     23     0%   1-103
aredis_om/model/render_tree.py              33     31     6%   24-75
aredis_om/model/token_escaper.py            13      1    92%   16
aredis_om/unasync_util.py                   20      7    65%   14, 18, 28, 32-34, 41
----------------------------------------------------------------------
TOTAL                                     1154    257    78%

==================================== short test summary info =====================================
FAILED tests_sync/test_hash_model.py::test_full_text_search_queries - AssertionError: assert [M...
FAILED tests_sync/test_hash_model.py::test_numeric_queries - AssertionError: assert [Member(pk=...
================================= 2 failed, 120 passed in 3.75s ==================================
make: *** [Makefile:66: test] Error 1
```

This might be due to the async nature of the tests, I was wondering if a failure like this one is because the async fixture hasn't fully created when the test is run and all 3 entries aren't yet saved in Redis... but I've read the [pytest-asyncio documentation](https://pypi.org/project/pytest-asyncio/) and believe the code in the test to be correct.

```bash
_________________________________________ test_all_keys __________________________________________
[gw1] darwin -- Python 3.9.5 /Users/simonprickett/Library/Caches/pypoetry/virtualenvs/redis-om-dXPLVk-w-py3.9/bin/python

members = (Member(pk='01FS7N05Y5XG9RKS3SE1HF8DFY', first_name='Andrew', last_name='Brookins', email='a@example.com', join_date=d...SQPHM', first_name='Andrew', last_name='Smith', email='as@example.com', join_date=datetime.date(2022, 1, 12), age=100))
m = Models(BaseHashModel=<class 'tests_sync.test_oss_redis_features.m.<locals>.BaseHashModel'>, Order=<class 'tests_sync.test_oss_redis_features.m.<locals>.Order'>, Member=<class 'tests_sync.test_oss_redis_features.m.<locals>.Member'>)

    @pytest.mark.asyncio
    def test_all_keys(members, m):
        pks = sorted([pk for pk in m.Member.all_pks()])
>       assert len(pks) == 3
E       assert 0 == 3
E         +0
E         -3

tests_sync/test_oss_redis_features.py:80: AssertionError
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -37,11 +37,18 @@ def key_prefix(request, redis):
 
 @pytest.fixture(scope="session", autouse=True)
 def cleanup_keys(request):
-    def cleanup_keys():
-        # Always use the sync Redis connection with finalizer. Setting up an
-        # async finalizer should work, but I'm not suer how yet!
-        from redis_om.connections import get_redis_connection as get_sync_redis
+    # Always use the sync Redis connection with finalizer. Setting up an
+    # async finalizer should work, but I'm not suer how yet!
+    from redis_om.connections import get_redis_connection as get_sync_redis
 
-        _delete_test_keys(TEST_PREFIX, get_sync_redis())
+    # Increment for every pytest-xdist worker
+    redis = get_sync_redis()
+    once_key = f"{TEST_PREFIX}:cleanup_keys"
+    redis.incr(once_key)
 
-    request.addfinalizer(cleanup_keys)
+    yield
+
+    # Delete keys only once
+    if redis.decr(once_key) == 0:
+        _delete_test_keys(TEST_PREFIX, redis)
+        redis.delete(once_key)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -51,4 +51,3 @@ def cleanup_keys(request):
     # Delete keys only once
     if conn.decr(once_key) == 0:
         _delete_test_keys(TEST_PREFIX, conn)
-        conn.delete(once_key)

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -42,13 +42,13 @@ def cleanup_keys(request):
     from redis_om.connections import get_redis_connection as get_sync_redis
 
     # Increment for every pytest-xdist worker
-    redis = get_sync_redis()
+    conn = get_sync_redis()
     once_key = f"{TEST_PREFIX}:cleanup_keys"
-    redis.incr(once_key)
+    conn.incr(once_key)
 
     yield
 
     # Delete keys only once
-    if redis.decr(once_key) == 0:
-        _delete_test_keys(TEST_PREFIX, redis)
-        redis.delete(once_key)
+    if conn.decr(once_key) == 0:
+        _delete_test_keys(TEST_PREFIX, conn)
+        conn.delete(once_key)

--- a/tests/test_hash_model.py
+++ b/tests/test_hash_model.py
@@ -137,7 +137,7 @@ async def test_full_text_search_queries(members, m):
 
     assert actual == [member1]
 
-    actual = await (m.Member.find(~(m.Member.bio % "anxious")).sort_by('age').all())
+    actual = await (m.Member.find(~(m.Member.bio % "anxious")).sort_by("age").all())
 
     assert actual == [member1, member3]
 

--- a/tests/test_hash_model.py
+++ b/tests/test_hash_model.py
@@ -137,7 +137,7 @@ async def test_full_text_search_queries(members, m):
 
     assert actual == [member1]
 
-    actual = await (m.Member.find(~(m.Member.bio % "anxious")).all())
+    actual = await (m.Member.find(~(m.Member.bio % "anxious")).sort_by('age').all())
 
     assert actual == [member1, member3]
 
@@ -433,7 +433,7 @@ async def test_all_pks(m):
         bio="This is a test user to be deleted.",
     )
 
-    await member1.save()   
+    await member1.save()
 
     pk_list = []
     async for pk in await m.Member.all_pks():

